Pynite/Analysis.py

```diff
@@ -524,6 +528,25 @@ def _sum_displacements(model: FEModel3D, Delta_D1: NDArray[float64], Delta_D2: N
         node.RZ[combo.name] += Delta_D[node.ID*6 + 5, 0]
 
 
+def _expand_displacements(model, D1, D2, D1_indices, D2_indices):
+    """
+    Expands partitioned displacement vectors back to full DOF set.
+    Used for mode shape expansion.
+    """
+    total_dof = len(model.nodes) * 6
+    D_full = zeros((total_dof, 1))
+    
+    # Place D1 values (unknown DOFs)
+    for i, index in enumerate(D1_indices):
+        D_full[index] = D1[i]
+    
+    # Place D2 values (known DOFs - typically zeros for modal analysis)
+    for i, index in enumerate(D2_indices):
+        D_full[index] = D2[i]
+    
+    return D_full
+
```

Comment:
Redundant: There is already a function in Analysis.py that does this. Have a look at _unpartition_disp.